### DISCUSSION TITLE
add  support communication between processes in a machine to akka remote

### DISCUSSION THREAD

**Discussion by qthanpt** [akkadotnet/akka.net#7187] [Ideas] [answered]
hi
I want to develop application using akka.net & have comunication between processes in machine , but Akka Remote only support with TCP/gRPC with a port socket  each process. So I need use many port socket in my machine. You can help provide solution to communcation  remote actor in machine (e.g : NamePipe,...)

**Comment by GrayGreenGoose** [comment, marked as answer]
I was thinking about the same. 
Such thing seems to not be mostly expected, but would be very useful, because Akka.Net is great for just communication between processes and therefore for developing modular systems which are not necessarily working on different machines or over network.

In terms of protocol it could be either "named pipes" (even MS SQL uses that) or reading/writing data to a buffer created as "shared memory". Maybe there are also other approaches.
There is one more thing - I don't know how Linux does IPC, so there is a risk that the solution would be only for Windows platform.

**Reply by Aaronontheweb** [reply]
Maybe @DSanchen has some ideas on how to implement a generalizable NamedPipe transport for Windows - since that was part of the issue we just fixed here https://github.com/akkadotnet/akka.net/issues/7378

**Reply by Aaronontheweb** [reply]
cc @qthanpt

**Reply by DSanchen** [reply]
I implemented a rather basic local-only transport and created a repo at https://github.com/DSanchen/Akka.Remote.LocalOnlyIPC

Feel free to try it and leave comments on it. It is right now not meant to be production ready !

**Reply by MacKey-255** [reply]
Good job @DSanchen I'm working in a IPC using shared memory but I have permission problems with lock semaphores and some internal stream files load in memory (UnauthorizedAccessException) :(

**Comment by DSanchen** [comment]
I try to come up with a suggestion and throw it in here for further discussion...
